src/Microsoft.ML.Core/CommandLine/SpecialPurpose.cs

```diff
@@ -4,7 +4,8 @@
 
 namespace Microsoft.ML.Runtime.CommandLine
 {
-    public static class SpecialPurpose
+    [BestFriend]
+    internal static class SpecialPurpose
```

Comment:
>internal [](start = 4, length = 8)

maybe a one line comment about what this is used, like you did for the other ones.

Reply:
I don't know, you mean the other classes in this folder that I bore out of `CmdParser.cs`? I didn't actually add any documentation to them, if they had documentation before I got there, then I copied it out with the class, but this one I guess didn't have any. If it's all the same to you I'll just leave it as it was since I wasn't quite doing a documentation pass, and since it's internal code I'm not certain I see the urgency? Though in principle it should be documented if this is some concept we want to keep...